internal/commands/user/delete_test.go

```diff
@@ -0,0 +1,309 @@
+					Project: projectID,
+					App:     appID,
+				},
+				usersInputs: usersInputs{
+					Users: []string{testUsers[0].ID},
+				},
+			},
+			realmClient: realmClient,
+		}
+
+		assert.Nil(t, cmd.Handler(nil, nil))
+		assert.Equal(t, realm.AppFilter{App: appID, GroupID: projectID}, capturedAppFilter)
+		assert.Equal(t, projectID, capturedProjectID)
+		assert.Equal(t, appID, capturedAppID)
+		assert.Equal(t, testUsers[0].ID, cmd.inputs.Users[0])
+		assert.Equal(t, cmd.outputs[0].err, errors.New("client error"))
+		assert.Equal(t, cmd.outputs[0].user, testUsers[0])
+	})
+
+	t.Run("should return an error", func(t *testing.T) {
+		for _, tc := range []struct {
+			description string
+			setupClient func() realm.Client
+			expectedErr error
+		}{
+			{
+				description: "when resolving the app fails",
+				setupClient: func() realm.Client {
+					realmClient := mock.RealmClient{}
+					realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+						return nil, errors.New("something bad happened")
+					}
+					return realmClient
+				},
+				expectedErr: errors.New("something bad happened"),
+			},
+			{
+				description: "when finding the users fails",
+				setupClient: func() realm.Client {
+					realmClient := mock.RealmClient{}
+					realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+						return []realm.App{testApp}, nil
+					}
+					realmClient.FindUsersFn = func(groupID, appID string, filter realm.UserFilter) ([]realm.User, error) {
+						return nil, errors.New("something bad happened")
+					}
+					return realmClient
+				},
+				expectedErr: errors.New("something bad happened"),
+			},
+		} {
+			t.Run(tc.description, func(t *testing.T) {
+				realmClient := tc.setupClient()
+
+				cmd := &CommandDelete{
+					realmClient: realmClient,
+				}
+
+				err := cmd.Handler(nil, nil)
+				assert.Equal(t, tc.expectedErr, err)
+			})
+		}
+	})
+}
+
+func TestUserDeleteFeedback(t *testing.T) {
+	testUsers := []realm.User{
+		{
+			ID: "user-1",
+			Identities: []realm.UserIdentity{
+				{ProviderType: providerTypeLocalUserPass},
+			},
+			Data:     map[string]interface{}{"email": "user-1@test.com"},
+			Disabled: false,
+			Type:     "type-1",
+		},
+	}
+	for _, tc := range []struct {
+		description    string
+		outputs        []userOutput
+		expectedOutput string
+	}{
+		{
+			description:    "should show no users to delete",
+			expectedOutput: "01:23:45 UTC INFO  No users to delete\n",
+		},
+		{
+			description: "should show 1 failed user",
+			outputs: []userOutput{
+				{user: testUsers[0], err: errors.New("client error")},
+			},
+			expectedOutput: strings.Join(
+				[]string{
+					"01:23:45 UTC INFO  Provider type: local-userpass",
+					"  Email            ID      Type    Deleted  Details     ",
+					"  ---------------  ------  ------  -------  ------------",
+					"  user-1@test.com  user-1  type-1  false    client error",
+					"",
+				},
+				"\n",
+			),
+		},
```

Comment:
can we have a "many" test case as well with multiple provider types, some error'd some not?

Reply:
done, see below